addons/block_code/code_generation/blocks_catalog.gd

```diff
@@ -27,3 +34,10 @@ static func get_block(block_name: StringName):
 
 static func has_block(block_name: StringName):
 	return block_name in _catalog
+
+
+static func get_blocks_by_class(_class_name: String):
+	if not _class_name in _by_class_name:
+		return []
+	var block_definitions = _by_class_name[_class_name] as Dictionary
+	return block_definitions.values()
```

Comment:
Unfortunately no. See 028fa109 how I ported the custom blocks. Do you think they should be defined in resource files and added automatically to the catalog?

Reply:
Yeah, that made sense to me once I got to the later commits :) I think that would be very clever if they were defined in resource files, but I can see how that's a tangent from the purpose of this branch. I'm tempted to play with that in another branch after I've done some more pressing things. A start would be to make `BlocksCatalog._BLOCKS_PATH` a list and solve from there.

Reply:
@dylanmccall you made me think that the [ResourceLoader](https://docs.godotengine.org/en/stable/classes/class_resourceloader.html#class-resourceloader-method-add-resource-format-loader) is already loading these resource files, so we are actually loading them twice using DirAccess. In fact, the DirAccess docs say "use ResourceLoader to access imported resources". So maybe a better idea would be:
- Add a custom extension to these resource files like `.block` or `.blockdef`.
- Add a ResourceFormatLoader that can handle that extension and add the block definitions to the catalog as they load.

Then custom resource files can be added to the user project without caring about the filesystem path.

Reply:
If we had a custom definition it would also help later for translation since we likely [need a custom translation parser](https://docs.godotengine.org/en/stable/classes/class_editortranslationparserplugin.html) that matches by file extension. That said, I think we should punt for the moment and try to get this work landed.